plasmapy/analytic/parameters.py

```diff
@@ -0,0 +1,852 @@
+    function.  However, the definition of thermal velocity varies by
+    the square root of two depending on whether or not this velocity
+    absorbs that factor in the expression for a Maxwellian
+    distribution.  In particular, the expression given in the NRL
+    Plasma Formulary [1] is a square root of two smaller than the
+    result from this function.
+
+    Examples
+    --------
+    >>> ion_thermal_speed(5*u.eV)
+    <Quantity 30949.690763378258 m / s>
+    >>> ion_thermal_speed(1e6*u.K, ion='p')
+    <Quantity 128486.56960876317 m / s>
+
+    """
+
+    try:
+        T_i = T_i.to(u.K, equivalencies=u.temperature_energy())
+        m_i = ion_mass(ion)
+        V_Ti = (np.sqrt(2*k_B*T_i/m_i)).to(u.m/u.s)
+    except Exception:
+        raise ValueError("Unable to find ion thermal speed")
+
+    if V_Ti > c:
+        raise UserWarning("Ion thermal speed is greater than speed of light")
+
+    return V_Ti
+
+
+@u.quantity_input
+def electron_gyrofrequency(B: u.T):
+    """Calculate the electron gyrofrequency in units of radians per second.
+
+    Parameters
+    ----------
+    B: Quantity
+        Magnetic field strength
+
+    Returns
+    -------
+    omega_ce: Quantity
+        Electron gyrofrequency in radians per second.
+
+    Notes
+    -----
+    The electron gyrofrequency is the angular frequency of electrons
+    gyration around magnetic field lines.
+
+    The electron gyrofrequency is also known as the electron cyclotron
+    frequency or the electron Larmor frequency.
+
+    The recommended way to convert from angular frequency to frequency
+    is to use an equivalency between cycles per second and Hertz, as
+    Astropy's dimensionles_angles() equivalency does not account for
+    the factor of 2*pi needed during this conversion.  The
+    dimensionless_angles() equivalency is appropriate when dividing a
+    velocity by an angular frequency to get a length scale.
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> from numpy import pi
+    >>> omega_ce = electron_gyrofrequency(0.1*u.T)
+    >>> print(omega_ce)
+    17588200878.472023 rad / s
+    >>> f_ce = omega_ce.to(u.Hz, equivalencies=[(u.cy/u.s, u.Hz)])
+    >>> print(f_ce)
+    2799249109.9020386 Hz
+
+    """
+
+    try:
+        omega_ce = u.rad*(e*B/m_e).to(1/u.s)
+    except Exception:
+        raise ValueError("Unable to find electron gyrofrequency")
+
+    return omega_ce
+
+
+def ion_gyrofrequency(B: u.T, ion='p', Z=None):
+    """Calculate the ion gyrofrequency in units of radians per second.
+
+    Parameters
+    ----------
+    B: Quantity
+        Magnetic field strength
+    ion : string
+        Symbol representing the ion species, defaulting to protons
+
+    Returns
+    -------
+    omega_ci: Quantity
+        The ion gyrofrequency in units of radians per second
+
+    Notes
+    -----
+    The ion gyrofrequency is the angular frequency of ion gyration
+    around magnetic field lines.
```

Comment:
```markdown
and is given by:

 .. math::
 omega_{ce} = \frac{e B}{m_i}
```

I know these things are obvious to most, but hey, this was supposed to be readable for **undergrads**. ;)